### PR TITLE
Fix for 1076

### PR DESCRIPTION
**Pull request by sbrossie**

### CODE REVIEW

File: payment/src/test/java/org/killbill/billing/payment/core/TestPaymentMethodProcessorWithDB.java [hunk 1]
```diff
@@ -99,7 +99,6 @@ public void testDeletePaymentMethod() throws Exception {
         Assert.assertEquals(history2.getAccountRecordId(), paymentMethodModelDao.getAccountRecordId());
         Assert.assertEquals(history2.getTenantRecordId(), paymentMethodModelDao.getTenantRecordId());
         Assert.assertEquals(history2.getExternalKey(), paymentMethodModelDao.getExternalKey());
-        // Note: upon deletion, the recorded state is the same as before the delete
```
Comment (pierre) [inline payment/src/test/java/org/killbill/billing/payment/core/TestPaymentMethodProcessorWithDB.java]: See, we knew about it! ;-)

File: util/src/main/java/org/killbill/billing/util/entity/dao/EntitySqlDaoWrapperInvocationHandler.java [hunk 2]
```diff
@@ -276,7 +258,7 @@ public Object execute() throws Throwable {
             }
         } else {
             // For updates, easiest is to go back to the database
```
Comment (pierre) [inline util/src/main/java/org/killbill/billing/util/entity/dao/EntitySqlDaoWrapperInvocationHandler.java]: Nit: comment should be updated.

Comment (sbrossie) [reply util/src/main/java/org/killbill/billing/util/entity/dao/EntitySqlDaoWrapperInvocationHandler.java]: Ack, will do.

File: payment/src/test/java/org/killbill/billing/payment/core/TestPaymentMethodProcessorWithDB.java (hunk 1 above)
Comment (sbrossie) [reply payment/src/test/java/org/killbill/billing/payment/core/TestPaymentMethodProcessorWithDB.java]: If by 'we' you mean the common pool of knowledge that we both have, then 'yes' !

**Pull request merged**